tokio/src/macros/select.rs

```diff
@@ -608,10 +608,24 @@ macro_rules! select {
 
     // ===== Entry point =====
 
+    (biased; else => $else:expr $(,)? ) => {
+        $crate::select!{
+            biased;
+            _ = async {}, if false => unreachable!(),
+            else => $else }
+    };
```

Comment:
It looks like we can just expand this to `$else` directly.

Reply:
I did it now, I merged the 2 new rules into one and am using `$else` directly.

```rust
    ($(biased;)? else => $else:expr $(,)? ) => {
        $crate::select!(@{ start=0; (_) () _ = async {}, if false => unreachable!(), ; $else })
    };
```

Reply:
That's not what I meant. What I meant was to just expand the macro to `{$else}` when there's only an else branch, instead of expanding to something with an `if false` branch.

Reply:
Oh I see, it's better indeed !
I don't know why I didn't see this obvious simple solution !
I updated it.

Reply:
@Darksonn I am having a issue with Git.... I applied your advice, and now I wanted to "squash" the 3 commits into one so I can give you a proper nice PR with one commit, but it seems i failed to do so.... (I struggle with Git manipulations...)

Is there a simple fool proof way to do that so my PR can be better next time?

Reply:
I will squash them when I merge them, so don't worry about that.

But to answer your question, my favorite tool for something like this is _interactive rebase_. In your case, you would call `git rebase -i 84e41d4affe` (here `84e41d4affe` is the commit before the thing you want to change, which you can get from `git log`). Then you can mark everything but your first commit as "fixup" (replace the word "pick"). When you execute that rebase, it will squash the fixups into the first commit. Afterwards, you can use `git commit --amend` to change the commit message of the squashed commit.